packages/pdf/src/RPA/PDF/keywords/document.py

```diff
@@ -0,0 +1,510 @@
+import os
+import tempfile
+from pathlib import Path
+from typing import Any
+
+import pdfminer
+import PyPDF2
+from fpdf import FPDF, HTMLMixin
+from pdfminer.pdfdocument import PDFDocument
+from pdfminer.pdfparser import PDFParser
+from PIL import Image
+
+from RPA.core.helpers import required_param
+from RPA.core.notebook import notebook_print
+from RPA.PDF.keywords import (
+    LibraryContext,
+    keyword,
+)
+
+
+class PDF(FPDF, HTMLMixin):
+    pass
+
+
+class DocumentKeywords(LibraryContext):
+    """Keywords for basic PDF operations"""
+
+    def __init__(self, ctx):
+        super().__init__(ctx)
+        self.fpdf = PDF()
+        self._output_directory = Path(".")
+
+    @property
+    def output_directory(self):
+        return self._output_directory
+
+    @output_directory.setter
+    def output_directory(self, path: str):
+        self.output_directory = Path(path)
+
+    def __del__(self):
+        self.close_all_pdf_documents()
+
+    def close_all_pdf_documents(self) -> None:
+        """Close all opened PDF file descriptors."""
+        for filename, fileobject in self.ctx.fileobjects.items():
+            fileobject.close()
+            self.logger.debug('PDF "%s" closed', filename)
+        self.ctx.anchor_element = None
+        self.ctx.fileobjects = {}
+        self.ctx.active_pdf = None
+        self.ctx.active_fileobject = None
+        self.ctx.active_fields = None
+        self.ctx.rpa_pdf_document = None
+
+    @keyword
+    def open_pdf_document(self, source_pdf: str = None) -> None:
+        """Open PDF document.
+
+        :param source_pdf: filepath to the source pdf
+        :raises ValueError: if PDF is already open
+
+        Also opens file for reading.
+        """
+        if source_pdf is None:
+            raise ValueError("Source PDF is missing")
+        if str(source_pdf) in self.ctx.fileobjects.keys():
+            raise ValueError(
+                "PDF file is already open. Please close it before opening again."
+            )
+        self.ctx.active_pdf = str(source_pdf)
+        self.ctx.active_fileobject = open(source_pdf, "rb")
+        self.ctx.active_fields = None
```

Comment:
should the active document be some sort of container for these? keeping the path and handle in sync seems to be a bit of a hassle.